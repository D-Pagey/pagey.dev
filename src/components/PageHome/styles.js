import styled from 'styled-components/macro';

export const Wrapper = styled.div`
    align-items: center;
    display: flex;
    flex-direction: column;
    padding: 2rem 0 0;
`;

export const IntroGrid = styled.div`
    display: grid;
    max-width: 365px;

    @media (min-width: 600px) {
        grid-column-gap: 1rem;
        grid-template-columns: 1fr 1fr;
        grid-template-rows: max-content;
        max-width: none;
    }
`;

export const Image = styled.img`
    border-radius: 3px;
    margin: 0 auto;
    width: 325px;

    @media (min-width: 600px) {
        grid-row: 1 / 3;
    }
`;

export const Text = styled.p`
    margin: 1rem 0 0;
`;
