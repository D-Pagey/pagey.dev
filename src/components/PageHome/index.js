import React from 'react';
import ShowcaseList from '../ShowcaseList';
import pagey from './assets/pagey.jpg';
import * as S from './styles';

export const PageHome = () => (
    <S.Wrapper data-testid="pageHome">
        <S.IntroGrid>
            <S.Image src={pagey} alt="pagey" />

            <S.Text>
                Yo, my name is Dan, I&apos;m a JavaScript Developer and I enjoy building apps to solve problems.
            </S.Text>

            <S.Text>
                I have always appreciated well done data visualisation and so I&apos;m always looking for an excuse to
                use raw SVGs and animate them. In my spare time, I am often either building a side project or looking up
                some cool new technology that I will likely never use.
            </S.Text>
        </S.IntroGrid>

        <ShowcaseList />
    </S.Wrapper>
);
