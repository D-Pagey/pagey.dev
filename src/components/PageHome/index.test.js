import React from 'react';
import mockAxios from 'axios';
import PageHome from '.';

describe('PageHome component', () => {
    beforeEach(() => {
        mockAxios.get.mockImplementationOnce(() =>
            Promise.resolve({
                data: {
                    html_url: 'github.com/namo',
                    description: 'Namo is an app...',
                    name: 'namo',
                    pushed_at: '2019-10-16T11:20:56Z',
                    homepage: 'https://namo.london'
                }
            })
        );
        mockAxios.get.mockImplementationOnce(() =>
            Promise.resolve({
                data: {
                    html_url: 'github.com/heidan',
                    description: 'Heidan is an app...',
                    name: 'heidan',
                    pushed_at: '2019-10-16T11:20:56Z',
                    homepage: 'https://heidan.london'
                }
            })
        );
        mockAxios.get.mockImplementationOnce(() =>
            Promise.resolve({
                data: {
                    html_url: 'github.com/pantry',
                    description: 'Pantry is an app...',
                    name: 'pantry',
                    pushed_at: '2019-10-16T11:20:56Z',
                    homepage: 'https://pantry.london'
                }
            })
        );
    });

    it('should render', () => {
        const { container } = render(<PageHome />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
