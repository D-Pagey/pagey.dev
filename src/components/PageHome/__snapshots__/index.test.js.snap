// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHome component should render 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 2rem 0 0;
}

.c1 {
  display: grid;
  max-width: 365px;
}

.c2 {
  border-radius: 3px;
  margin: 0 auto;
  width: 325px;
}

.c3 {
  margin: 1rem 0 0;
}

@media (min-width:600px) {
  .c1 {
    grid-column-gap: 1rem;
    grid-template-columns: 1fr 1fr;
    grid-template-rows: max-content;
    max-width: none;
  }
}

@media (min-width:600px) {
  .c2 {
    grid-row: 1 / 3;
  }
}

<div
  class="c0"
  data-testid="pageHome"
>
  <div
    class="c1"
  >
    <img
      alt="pagey"
      class="c2"
      src="pagey.jpg"
    />
    <p
      class="c3"
    >
      Yo, my name is Dan, I'm a JavaScript Developer and I enjoy building apps to solve problems.
    </p>
    <p
      class="c3"
    >
      I have always appreciated a good data vizualisation and so I'm learning to use D3.js within React apps. Hopefully, I can build something cool and interactive. Until then, this is my playground.
    </p>
  </div>
</div>
`;
